portfolio/src/main/java/com/google/sps/servlets/LoginServlet.java

```diff
@@ -0,0 +1,52 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/login")
+public class LoginServlet extends HttpServlet {
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    response.setContentType("text");
+
+    UserService userService = UserServiceFactory.getUserService();
+    if (userService.isUserLoggedIn()) {
+      response.getWriter().println("yes");
+    } else {
+      response.getWriter().println("no");
```

Comment:
Not sure if you're planning to change this in an upcoming PR, but I'd recommend against hard-coding in "yes" or "no" values as plain text for whether or not the user is logged in. Consider using HTML to show different elements depending on if the user is logged in or not, or possibly using a basic JSON object to store the "isLoggedIn" value.

Reply:
That is a good point, thank you! I made a change and directly stored the boolean value of userService.isUserLoggedIn(), but then passed it to js in the response as a string. I think this is better than hard-coding yes or no, but I'm not sure if it's quite what you meant. If it would be even better to store it as a JSON object let me know! I tried doing that for a little but was running into some issues, but I can definitely spend more time on that if my changes are still not great code.